xarray_beam/_src/core.py

```diff
@@ -304,3 +304,32 @@ def expand(self, pcoll):
             self._key_to_chunks, num_threads=self.num_threads
         )
     )
+
+
+def validate_chunk(key: Key, dataset: xarray.Dataset) -> None:
+  """Verify that keys correpond to Dataset properties."""
+  missing_keys = [f"'{k}'" for k in key.offsets.keys() if k not in dataset.dims]
+  if missing_keys:
+    raise ValueError(
+      f"Key offset(s) {', '.join(missing_keys)} not found in Dataset dimensions."
```

Comment:
nit: this should have two extra spaces of indent per google style guide.

Reply:
I would also suggest printing the offending `key` and `dataset` in the error message.